sydent/db/migration.py

```diff
@@ -0,0 +1,182 @@
+import json
+import sqlite3
+
+import signedjson.sign
+
+from sydent.util import json_decoder
+from sydent.util.emailutils import sendEmail
+from sydent.util.hash import sha256_and_url_safe_base64
+
+
+def update_assosc(self, conn: sqlite3.Connection):
+    # from table local_threepid_associations, select all associations where the medium is 'email'
+    cur = conn.cursor()
+
+    res = cur.execute(
+        "SELECT address, mxid FROM local_threepid_associations WHERE medium = 'email'"
+        "ORDER BY ts DESC"
+    )
```

Comment:
So this I don't know-I do know that the deletion step is supposed to be rare, but as for how large a DB this code will run on, I have no idea-this might be a questions for Brendan.